### PR TITLE
Rework EffBroadcast to support advanced formatting

### PR DESCRIPTION
**Pull request by APickledWalrus**
### Description
This PR reworks EffBroadcast to support advanced formatting by using similar handling to EffMessage.
(I know this is a fixes branch but this is actually an enhancement as it was intended behavior)

---
**Target Minecraft Versions:** Any
**Requirements:** None
**Related Issues:** #4156

### CODE REVIEW

File: src/main/java/ch/njol/skript/effects/EffBroadcast.java [hunk 1]
```diff
@@ -57,30 +63,37 @@
...
-						player.sendMessage(string);
-					}
+		if (messages instanceof ExprColoured && ((ExprColoured) messages).isUnsafeFormat()) { // Manually marked as trusted
+			for (Object message : messages.getArray(e)) {
+				BaseComponent[] components = BungeeConverter.convert(ChatMessages.parse((String) message));
+				receivers.forEach(receiver -> receiver.spigot().sendMessage(components));
+			}
+		} else {
+			for (Object message : messages.getArray(e)) {
+				if (message instanceof VariableString) {
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/effects/EffBroadcast.java:91]: VariableString is an expression, and expression's getter (e.g. `message.getArray(e)`) will not return another expression

File: src/main/java/ch/njol/skript/effects/EffBroadcast.java [hunk 2]
```diff
@@ -18,32 +18,38 @@
...
+import java.util.List;
 
 @Name("Broadcast")
-@Description({"Broadcasts a message to the server. Only formatting options supported by console",
-		"(i.e. colors) are supported. If you need to use advanced chat formatting, send the",
-		"message to all players instead of broadcasting it."})
-@Examples({"broadcast \"Welcome %player% to the server!\"",
-		"broadcast \"Woah! It's a message!\""})
-@Since("1.0, 2.6 (broadcasting objects)")
+@Description("Broadcasts a message to the server. Only formatting options supported by console (i.e. colors) are supported.")
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/effects/EffBroadcast.java:47]:
```suggestion
@Description("Broadcasts a message to the server.")
```

**Review by TPGamesNL — CHANGES_REQUESTED** [review]
Looks good!
What is sent to console when you use advanced formatting?

**Comment by APickledWalrus** [comment]
> What is sent to console when you use advanced formatting?

image
The advanced stuff seems to be ignored :)

**Review by TPGamesNL — APPROVED** [review]

**Pull request merged**